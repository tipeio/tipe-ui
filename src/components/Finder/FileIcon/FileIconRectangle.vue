<template>
  <div :data-tipe-ui="$options.name">
    <button
      class="file-icon"
      @click="$emit('click', $event)"
    >
      <tipe-icon :icon="icon" />
    </button>
    <button
      class="label"
      @click="$emit('click', $event)"
    >
      {{ file.label }}
    </button>
    <button
      class="options-icon"
    >
      <tipe-icon icon="options" />
    </button>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import interfaces from '@tipe/tipe-interfaces'
import TipeIcon from '@/components/Icon'
import { getFileIcon } from './utils'

export default {
  name: 'TipeFileIconRectangle',
  components: {
    TipeIcon
  },
  props: {
    file: vueTypes.shape(interfaces.file)
  },
  computed: {
    icon() {
      return getFileIcon(this.file)
    }
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeFileIconRectangle'] {
  height: 100%;
  width: 100%;
  min-width: 9rem;
  min-height: 3rem;
  display: grid;
  grid-template-columns: 3.75rem 1fr 3rem;
  border: 1px solid var(--gray-blue-light);
  border-radius: 0.3125rem;
  box-sizing: border-box;
}

button {
  border: none;
  outline: none;
  margin: 0;
  padding: 0;
  background-color: transparent;
}

.file-icon,
.options-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}

.label {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  white-space: nowrap;
  overflow: hidden;
  color: rgb(31, 52, 108);
  font-size: 0.8125rem;
}
</style>
