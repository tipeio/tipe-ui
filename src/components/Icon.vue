<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
