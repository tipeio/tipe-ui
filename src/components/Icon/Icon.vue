<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
