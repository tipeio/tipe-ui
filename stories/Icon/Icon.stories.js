import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
