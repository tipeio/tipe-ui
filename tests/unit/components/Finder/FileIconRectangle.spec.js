import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import FileIconRectangle from '@/components/Finder/FileIcon/FileIconRectangle'
import Icon from '@/components/Icon'
import { mocks, fixtures } from '@tipe/tipe-test-utils'

describe('FileIconRectangle', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const propsData = { file: fixtures.file() }
      const wrapper = shallowMount(FileIconRectangle, { propsData })

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const propsData = { file: mocks.file() }
      const wrapper = shallowMount(FileIconRectangle, { propsData })

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeFileIconRectangle')
    })
  })

  describe(':props', () => {
    it(':file - should render label prop as label', () => {
      const propsData = { file: mocks.file() }
      const wrapper = shallowMount(FileIconRectangle, { propsData })

      expect(wrapper.find('.label').text()).toEqual(propsData.file.label)
    })

    it(':type(folder) - should pass "folder" to Icon component\'s :icon prop', () => {
      const propsData = { file: mocks.file({ type: 'folder' }) }
      const wrapper = shallowMount(FileIconRectangle, { propsData })

      expect(wrapper.find(Icon).props().icon).toEqual('folder-filled')
    })

    it(':type(document) - should pass "file" to Icon component\'s :icon prop', () => {
      const propsData = { file: mocks.file({ type: 'document' }) }
      const wrapper = shallowMount(FileIconRectangle, { propsData })

      expect(wrapper.find(Icon).props().icon).toEqual('document-filled')
    })
  })

  describe('@events', () => {
    it('@click - emits', () => {
      const propsData = { file: mocks.file() }
      const wrapper = shallowMount(FileIconRectangle, { propsData })

      wrapper.find('.file-icon').trigger('click')
      expect(wrapper.emitted().click).toBeTruthy()

      wrapper.find('.label').trigger('click')
      expect(wrapper.emitted().click).toBeTruthy()
    })
  })
})
