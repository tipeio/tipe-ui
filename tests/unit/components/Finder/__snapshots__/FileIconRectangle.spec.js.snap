// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileIconRectangle <template> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeFileIconRectangle" file="[object Object]"><button class="file-icon">
    <TipeIcon-stub icon="" height="1rem" width="1rem" color="#000" __iconLoader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}"></TipeIcon-stub>
  </button> <button class="label">

  </button> <button class="options-icon">
    <TipeIcon-stub icon="options" height="1rem" width="1rem" color="#000" __iconLoader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}"></TipeIcon-stub>
  </button></div>
`;
